src/dialog-editor/components/field/fieldComponent.ts

```diff
@@ -38,6 +38,42 @@ class FieldController {
     this.DialogEditor.updatePositions(this.getFields(tabId, boxId));
   }
 
+  /**
+   * sort Field values if needed
+   * @memberof FieldController
+   * @function sortedField
+   * @returns {Array} of sorted field values
+   */
+  public sortedField() {
+    if (this.fieldData.options.sort_by !== 'none') {
+      let tempValues = [...this.fieldData.values];
+      if (this.fieldData.data_type === 'integer') {
+        tempValues.forEach((item, index) => {
+          const val = parseInt(tempValues[index][1], 10);
+          if (!Number.isNaN(val)) {
+            tempValues[index][1] = val;
```

Comment:
should `[1]` be `[sortBy]`?

Otherwise, you're always converting the label, but sometimes not sorting by label.